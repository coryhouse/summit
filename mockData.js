export const modelInputs = [
  { name: "name1", description: "desc1", date: "date1" },
  { name: "name2", description: "desc2", date: "date2" }
];
