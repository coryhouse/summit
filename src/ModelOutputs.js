import React from "react";

class ModelOutputs extends React.Component {
  render() {
    return (
      <div>
        <h2>Outputs</h2>
        <table>
          <thead>
            <tr>
              <th>Name</th>
              <th>Time</th>
              <th>Status</th>
              <th />
            </tr>
          </thead>
          <tbody>
            {this.props.modelInputs.map(modelInput => (
              <tr key={modelInput.id}>
                <td>{modelInput.name}</td>
                <td>{modelInput.description}</td>
                <td>{modelInput.date}</td>
                <td>
                  <button onClick={() => this.props.onDelete(modelInput.id)}>
                    Delete
                  </button>
                </td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}

export default ModelOutputs;
