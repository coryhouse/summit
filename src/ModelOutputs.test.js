// Add snapshot test here.
import React from "react";
import ModelOutputs from "./ModelOutputs";
import renderer from "react-test-renderer";
import { modelInputs as modelInputsMock } from "../mockData";

it("should render an empty table when passed an empty array", () => {
  // arrange
  const modelInputs = [];

  // act
  const tree = renderer
    .create(<ModelOutputs modelInputs={modelInputs} />)
    .toJSON();

  // assert
  expect(tree).toMatchSnapshot();
});

it("should render a table with 2 rows when passed two rows", () => {
  // arrange
  const modelInputs = modelInputsMock;

  // act
  const tree = renderer
    .create(<ModelOutputs modelInputs={modelInputs} />)
    .toJSON();

  // assert
  expect(tree).toMatchSnapshot();
});
