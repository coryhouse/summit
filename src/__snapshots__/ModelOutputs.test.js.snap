// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a table with 2 rows when passed two rows 1`] = `
<div>
  <h2>
    Outputs
  </h2>
  <table>
    <thead>
      <tr>
        <th>
          Input Name
        </th>
        <th>
          Time
        </th>
        <th>
          Status
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          name1
        </td>
        <td>
          desc1
        </td>
        <td>
          date1
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          name2
        </td>
        <td>
          desc2
        </td>
        <td>
          date2
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`should render an empty table when passed an empty array 1`] = `
<div>
  <h2>
    Outputs
  </h2>
  <table>
    <thead>
      <tr>
        <th>
          Input Name
        </th>
        <th>
          Time
        </th>
        <th>
          Status
        </th>
        <th />
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
